gst/nnstreamer/tensor_converter/tensor_converter.c

```diff
@@ -1049,61 +1050,45 @@ gst_tensor_converter_chain (GstPad * pad, GstObject * parent, GstBuffer * buf)
       frames_in = buf_size / frame_size;
       break;
     case _NNS_MEDIA_ANY:
+    {
+      GstTensorsConfig new_config;
       if (self->mode == _CONVERTER_MODE_CUSTOM_CODE) {
-        GstTensorsConfig new_config;
         if (self->custom.func == NULL) {
           nns_loge
               ("custom condition of the tensor_converter is not configured.");
           return GST_FLOW_ERROR;
         }
         inbuf = self->custom.func (buf, self->custom.data, &new_config);
-        if (inbuf == NULL) {
-          nns_loge ("Failed to run custom tensor converter.");
-          gst_buffer_unref (buf);
-          return GST_FLOW_ERROR;
-        }
-        if (gst_tensors_config_is_equal (config, &new_config) != TRUE) {
-          if (gst_tensor_converter_update_caps (self, self->srcpad,
-                  &new_config) == TRUE) {
-            self->tensors_config = new_config;
-          } else {
-            gst_buffer_unref (buf);
-            return GST_FLOW_ERROR;
-          }
-        }
         frames_in = 1;
         frame_size = gst_buffer_get_size (inbuf);
-
-        gst_buffer_unref (buf);
-        break;
       } else if (self->externalConverter && self->externalConverter->convert) {
-        GstTensorsConfig new_config;
-
-        inbuf =
-            self->externalConverter->convert (buf, &frame_size, &frames_in,
+        inbuf = self->externalConverter->convert (buf, &frame_size, &frames_in,
             &new_config);
-
-        if (gst_tensors_config_is_equal (config, &new_config) != TRUE) {
-          if (gst_tensor_converter_update_caps (self, self->srcpad,
-                  &new_config) == TRUE) {
-            self->tensors_config = new_config;
-          } else {
-            return GST_FLOW_ERROR;
-          }
-        }
-
-        g_assert (inbuf != NULL);
-        g_assert (frame_size > 0);
-
-        if (inbuf != buf)
-          gst_buffer_unref (buf);
-
-        break;
       } else {
         GST_ERROR_OBJECT (self, "Undefined behavior with type %d\n",
             self->in_media_type);
         return GST_FLOW_NOT_SUPPORTED;
       }
+
+      if (inbuf == NULL) {
+        nns_loge ("Failed to convert media to tensors.");
+        gst_buffer_unref (buf);
+        gst_tensors_info_free (&new_config.info);
+        return GST_FLOW_ERROR;
+      }
+
+      if (gst_tensors_config_is_equal (config, &new_config) != TRUE) {
```

Comment:
From https://developer.gnome.org/glib/stable/glib-Basic-Types.html
>>Never directly compare the contents of a gboolean variable with the values TRUE or FALSE. Use if (condition) to check a gboolean is "true", instead of if (condition == TRUE). Likewise use if (!condition) to check a gboolean is "false".

```suggestion
      if (!gst_tensors_config_is_equal (config, &new_config)) {
```